### DISCUSSION TITLE
Auto-trim functionality

### DISCUSSION THREAD

**Discussion by val3n71na** [fabricjs/fabric.js#9508] [Q&A] [answered]
Hello,

Is there a way to implement auto-trim functionality similar to what they have in AutoCAD? 

I have an SVG shape added to my fabric.js canvas using fabric.loadSVGfromString(). Then I add a circle over it using new fabric.Circle() and I need to be able to auto-trim the internal lines because I only need to have the outline of the shapes.

![image](https://github.com/fabricjs/fabric.js/assets/26135080/0041d3bc-009f-4c05-a2f5-8815a2790473)
From this ^ I need the auto trim to get the shape below:
![outline](https://github.com/fabricjs/fabric.js/assets/26135080/3f2b1b74-65bd-4ada-8b15-d384c4bdcb6a)

Do you have any idea if that is possible or what I should try?

Thank you!

**Comment by ShaMan123** [comment, marked as answer]
use a clipPath

**Reply by ShaMan123** [reply]
or globalCompositeOperation

**Comment by asturur** [comment]
@val3n71na i think paperjs or gojs do what you are asking
https://paperjs.org/examples/boolean-operations/
or
https://paperjs.org/examples/path-intersections/

maybe there is some useful code for you there.

**Reply by val3n71na** [reply]
thank you, asturur! I am going to check on it.
